packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useCallback, useContext } from 'react';
+import { useFeatureIsOn } from '@growthbook/growthbook-react';
+import { FeedPostClick } from './feed/useFeedOnPostClick';
+import { Post } from '../graphql/posts';
+import { Features } from '../lib/featureManagement';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+
+import { useExtensionPermission } from './useExtensionPermission';
+import { useLazyModal } from './useLazyModal';
+import { LazyModal } from '../components/modals/common/types';
+import { LazyModalType } from '../components/modals/common';
+
+type CompanionTriggerProps = {
+  post: Post;
+  index?: number;
+  row?: number;
+  column?: number;
+};
+
+type OpenArticle = (data?: CompanionTriggerProps) => () => Promise<void>;
+
+export type CompanionTrigger = {
+  onFeedArticleClick: FeedPostClick;
+  onPostArticleClick: PostClick;
+  activateCompanion: () => void;
+  openArticle: OpenArticle;
+  toggleOpen: (value: boolean, data?: CompanionTriggerProps) => void;
+  lazyModal: LazyModalType<LazyModal>;
+};
+
+type PostClick = (e: React.MouseEvent, data: { post: Post }) => Promise<void>;
+
+export default function useCompanionTrigger(
+  customPostClickHandler: (
+    post: Post,
+    index?: number,
+    row?: number,
+    column?: number,
+  ) => Promise<void>,
+): CompanionTrigger {
+  const isExtension = process.env.TARGET_BROWSER;
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { modal, closeModal, openModal } = useLazyModal();
+  const featureEnabled = useFeatureIsOn(
+    Features.EngagementLoopJuly2023Companion.toString(), // this feels very smelly, but otherwise complains because of the Features<string> type
+  );
+
+  const { requestContentScripts, useContentScriptStatus } =
+    useExtensionPermission({
+      origin: 'companion modal permission button',
+    });
+
+  const { contentScriptGranted } = useContentScriptStatus();
+
+  const articleClickHandler = useCallback(
+    async (
+      e: React.MouseEvent,
+      { post, index, row, column }: Partial<CompanionTriggerProps> = {},
+    ) => {
+      // the check whether the user is logged in is done on the GrowthBook side
+      // the feature flag is enabled only for logged-in users
+      // -- check that this is correct
+      console.log({ isExtension, contentScriptGranted });
+      if (!isExtension || (isExtension && contentScriptGranted)) {
+        await customPostClickHandler(post, index, row, column);
+      } else {
+        e.preventDefault();
+        // setData({ post, index, row, column });
```

Comment:
Not sure what this is for?

Reply:
leftover code from a different approach I took where I was storing the relevant data in a local state variable.